examples/unspecified_types/connector.py

```diff
@@ -1,39 +1,70 @@
+# See the Technical Reference documentation (https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update)
+# and the Best Practices documentation (https://fivetran.com/docs/connectors/connector-sdk/best-practices) for details.
+
+# Import required classes from fivetran_connector_sdk
 from fivetran_connector_sdk import Connector
+from fivetran_connector_sdk import Logging as log
 from fivetran_connector_sdk import Operations as op
 
 
+# Define the schema function which lets you configure the schema your connector delivers.
+# See the technical reference documentation for more details on the schema function:
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#schema
+# The schema function takes one parameter:
+# - configuration: a dictionary that holds the configuration settings for the connector.
 def schema(configuration: dict):
     return [
         {
-            "table": "unspecified",
-            "primary_key": ["id"],
+            "table": "unspecified",  # Name of the table in the destination.
+            "primary_key": ["id"],  # Primary key column(s) for the table.
+            # No columns are defined, meaning the types will be inferred.
         },
     ]
 
 
+# Define the update function, which is a required function, and is called by Fivetran during each sync.
+# See the technical reference documentation for more details on the update function:
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary containing any secrets or payloads you configure when deploying the connector.
+# - state: a dictionary containing the state checkpointed during the prior sync.
+#   The state dictionary is empty for the first sync or for any full re-sync.
 def update(configuration: dict, state: dict):
-    yield op.upsert(table="unspecified",
-                    data={
-                        "id": 1,
-                        "_bool": True,
-                        "_short": 15,
-                        "_long": 132353453453635,
-                        "_dec": "105.34",
-                        "_float": 10.4,
-                        "_double": 1e-4,
-                        "_ndate": "2007-12-03",
-                        "_ndatetime": "2007-12-03T10:15:30",
-                        "_utc": "2007-12-03T10:15:30.123Z",
-                        "_binary": b"\x00\x01\x02\x03",
-                        "_xml": "<tag>This is XML</tag>",
-                        "_str": "This is a string",
-                        "_json": {"a": 10},
-                        "_null": None
-                    })
+    # Yield an upsert operation to insert/update the row in the "unspecified" table.
+    # The data dictionary contains various data types.
+    # Since the schema does not specify column types, they will be inferred from the data.
+    log.fine("upserting to table 'unspecified'")
+    yield op.upsert(
```

Comment:
be clear where the column names will come from

Reply:
added a comment